docker-assets/auth-api/auth_api_service.rb

```diff
@@ -0,0 +1,55 @@
+require "sinatra/base"
+require "dbus"
+require "json"
+
+class AuthApiService < Sinatra::Base
+  DEFAULT_USER_ATTRIBUTES = %w(mail givenname sn displayname domainname).freeze
+
+  set :bind, "0.0.0.0"
+  set :port, ENV["HTTPD_AUTH_API_SERVICE_PORT"]
+
+  get "/api/user_attrs/:userid", :provides => 'json' do
+    attrs_needed = params[:attributes].nil? ? DEFAULT_USER_ATTRIBUTES : params[:attributes].split(',')
+
+    begin
+      ifp_interface = dbus_ifp_interface
+      user_attrs = ifp_interface.GetUserAttr(params[:userid], attrs_needed).first
```

Comment:
Not sure how this works under the covers, but please be sure there are no injection security concerns here...both values come from params which are user-manipulable.

Reply:
All seemed ok, I've tried with funky characters, ye-old redirects for shell, the & and such for URI, got either blocked early (badly formed URI), sinatra complaining on parameter or rejected from the infopipe library call. 

Regardless, I will ask the dbus/infopipe interface library devs.  This is specifically for org.freedesktop.sssd.infopipe.GetUserAttrs and .GetUserGroups.